pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -291,6 +291,16 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation) *gardencorev1
 			Fn:           flow.TaskFn(botanist.DeployKubeScheduler).RetryUntilTimeout(defaultInterval, defaultTimeout),
 			Dependencies: flow.NewTaskIDs(deploySecrets, waitUntilKubeAPIServerIsReady),
 		})
+		deployKonnectivityServer = g.Add(flow.Task{
+			Name:         "Deploying konnectivity-server",
+			Fn:           flow.TaskFn(botanist.DeployKonnectivityServer).RetryUntilTimeout(defaultInterval, defaultTimeout),
+			Dependencies: flow.NewTaskIDs(deploySecrets, waitUntilKubeAPIServerIsReady),
```

Comment:
Any reason why we have to wait for KAPI to be ready? Can we deploy the k-server simultaneously to the KAPI deployment?